src/configure/configure.ts

```diff
@@ -512,13 +514,13 @@ class Orchestrator {
             appropriatePipelines = remotePipelines;
         }
         const pipelineMap = this.getMapOfUniqueLabels(appropriatePipelines);
+        this.filterPipelinesByResourceType(pipelineMap);
```

Comment:
We are still getting list of all applicable templates and then filtering here. I think that is redundant. We should get list of templates.
We need to break repo analysis and listing templates in two parts, and then do only repo analysis in this case.

Reply:
Yes, we are getting all templatesand then filtering.
for example, for AKS scenario itself, there can be 2 templates - helm chart and otherwise. So, in that scenario, we'll just filter on the basis of target, and will let the user decide which path to go.

Reply:
Ah, makes sense. Then my other comment about using a different template service API is redundant

Reply:
> As of now, if there isn't any dockerFile, this would mean that the template service should not return AKS template and after the filterByResource, the templates to proceed will be 0 which will give the error that the language of the repo is not supported.

can you check this behavior with Pulkit? In case dockerfile is not found, we might still want to proceed by creating a workflow by taking a default dockerfile path. This is because we know that user wants to deploy to AKS and hence we  do not fail just because dockerfile is not found. May be used can create one before they do "commit and push"

Reply:
I was discussing this with @anuragc617 and he mentioned that if we use the template service filter API then it will return all AKS specific templates. This way we won't have a hard dependency on repo analysis. The templates will be shown to user even if they do not have dockerfile. If they select the template, then dockerfile path value will be the default value specified in template. This way we won't block user from creating workflow even if they don't have dockerfile. This behavior will then be consistent with AKS deployment center experience.

Reply:
I am okay to take this a separate PR so that we can deploy v1 first and unblock integration with AKS extension. This improvement can be taken in parallel.

Reply:
So, just to confirm, the flow will be :
1. Calling repo analysis
2. If the reource is already selected, calling template service filter API based on the target resource, else keeping the flow as it is which gets templates based on the repo analysis response
3. Further flow remains the same

Right?
Will pick this in another PR

Reply:
Yes 👍